packages/types/src/index.ts

```diff
@@ -36,146 +36,579 @@ export interface Dialog {
+ * app. Read our [guide to enabling interactivity](https://api.slack.com/messaging/interactivity/enabling).
+ *
+ * https://api.slack.com/reference/messaging/block-elements#button
+ */
 export interface Button extends Action {
   type: 'button';
+
+  /**
+   * A text object that defines the button's text. Can only be of type: plain_text.
+   * Maximum length for the text in this field is 75 characters.
+   */
   text: PlainTextElement;
+
+  /**
+   * The value to send along with the
+   * [interaction payload](https://api.slack.com/messaging/interactivity/enabling#understanding-payloads).
+   * Maximum length for this field is 75 characters.
+   */
   value?: string;
+
+  /**
+   * The value to send along with the
+   * [interaction payload](https://api.slack.com/messaging/interactivity/enabling#understanding-payloads).
+   * Maximum length for this field is 75 characters.
+   */
   url?: string;
 }
 
+/**
+ * This is like a cross between a button and a select menu - when a user clicks
+ * on this overflow button, they will be presented with a list of options to
+ * choose from. Unlike the select menu, there is no typeahead field, and the
+ * button always appears with an ellipsis ("…") rather than customisable text.
+ *
+ * As such, it is usually used if you want a more compact layout than a select
+ * menu, or to supply a list of less visually important actions after a row of
+ * buttons.
+ *
+ * To use interactive elements like this, you will need to make some changes
+ * to prepare your app. Read our
+ * [guide to enabling interactivity](https://api.slack.com/messaging/interactivity/enabling).
+ *
+ * https://api.slack.com/reference/messaging/block-elements#overflow
+ */
 export interface Overflow extends Action {
   type: 'overflow';
+
+  /**
+   * An array of option objects to display in the menu. Maximum number of options
+   * is 5, minimum is 2.
+   */
   options: Option[];
+
+  /**
+   * A plain_text only text object that defines the placeholder text shown on
+   * the menu. Maximum length for the text in this field is 150 characters.
+   */
+  placeholder: PlainTextElement;
 }
 
+/**
+ * An element which lets users easily select a date from a calendar style UI.
+ * Date picker elements can be used inside of section and actions blocks.
+ *
+ * To use interactive elements like this, you will need to make some changes
+ * to prepare your app. Read our guide to enabling interactivity.
+ *
+ * https://api.slack.com/reference/messaging/block-elements#datepicker
+ */
 export interface Datepicker extends Action {
   type: 'datepicker';
+
+  /**
+   * The initial date that is selected when the element is loaded. This
+   * should be in the format `YYYY-MM-DD`.
+   */
   initial_date?: string;
+
+  /**
+   * A plain_text only text object that defines the placeholder text shown
+   * on the menu. Maximum length for the text in this field is 150 characters.
+   */
   placeholder?: PlainTextElement;
 }
 
 /*
  * Block Types
  */
 
+/**
```

Comment:
I'd remove this since it's used above Block as well. This is really just a type helper.